pilot/pkg/bootstrap/server.go

```diff
@@ -935,6 +936,21 @@ func (s *Server) getCertKeyPaths(tlsOptions TLSOptions) (string, string) {
 
 // setPeerCertVerifier sets up a SPIFFE certificate verifier with the current istiod configuration.
 func (s *Server) setPeerCertVerifier(tlsOptions TLSOptions) error {
```

Comment:
I don't see how this would work in general. There are many many many cases where Istiod is not running in the same cluster as workloads. Determining authentication based on runtime environment is not going to work..

Reply:
You are right, the SpiffeBundleEndpoints part won't work. It is protocol based, not cloud provider based.